eNMS/services/configuration/netconf.py

```diff
@@ -0,0 +1,304 @@
+            ("None", "None"),
+        ),
+        label="Error option",
+        validate_choice=False,
+    )
+    lock = BooleanField(label="Lock target")
+    unlock = BooleanField(label="Unlock target")
+    copy_source = SelectField(
+        choices=(
+            ("running", "Running"),
+            ("candidate", "Candidate"),
+            ("startup", "Startup"),
+            ("srce_url", "Source URL"),
+        ),
+        label="Copy Source",
+        validate_choice=False,
+    )
+    source_url = StringField(
+        label="Copy source URL",
+        widget=TextArea(),
+        render_kw={"rows": 1},
+        substitution=True,
+    )
+    copy_destination = SelectField(
+        choices=(
+            ("running", "Running"),
+            ("candidate", "Candidate"),
+            ("startup", "Startup"),
+            ("dest_url", "Destination URL"),
+        ),
+        label="Copy Destination",
+        validate_choice=False,
+    )
+    destination_url = StringField(
+        label="Copy destination URL",
+        widget=TextArea(),
+        render_kw={"rows": 1},
+        substitution=True,
+    )
+    commit_conf = BooleanField(label="Commit")
+    timeout = IntegerField(default=15)
+    xml_conversion = BooleanField(
+        label="Convert XML result to dictionary", default=True
+    )
+    groups = {
+        "NETCONF Parameters": {
+            "commands": [
+                "nc_type",
+                "xml_filter",
+                "target",
+                "copy_source",
+                "source_url",
+                "copy_destination",
+                "destination_url",
+                "default_operation",
+                "test_option",
+                "error_option",
+                "lock",
+                "unlock",
+                "commit_conf",
+                "xml_conversion",
+            ],
+            "default": "expanded",
+        },
+        **ConnectionForm.groups,
+    }
+
+    # this hidden field is used to pass information to javascript so the field
+    # visibility can be changed as set below
+
+    input_data = HiddenField(
+        "",
+        default=dumps(
+            {
+                # add all fields to the field list except for "nc_type"
+                # which will drive the selection
+                # every field in this list will be hidden unless it is
+                # contained in one of the netconf type entries below.
+                "fields": [
+                    "xml_filter",
+                    "target",
+                    "copy_source",
+                    "source_url",
+                    "copy_destination",
+                    "destination_url",
+                    "default_operation",
+                    "test_option",
+                    "error_option",
+                    "lock",
+                    "unlock",
+                    "commit_conf",
+                    "xml_conversion",
+                ],
```

Comment:
that list seems redundant with the netconf type dictionary, isn't it the list of all values in `"netconf_type"` ? in which case you can create it in python directly (`list(set(sum(the_dict.values(), [])))` is one way to do it I guess) ?
if you compute it like that, you can use it above in the `groups` dictionary as well, so you can remove 20 lines of duplicated code